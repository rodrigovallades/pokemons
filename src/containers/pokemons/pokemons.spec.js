import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { Pokemons } from './index'

describe('Pokemons container', function () {
  beforeEach(function() {
    global.sessionStorage = jest.genMockFunction();
    global.sessionStorage.setItem = jest.genMockFunction();
    global.sessionStorage.getItem = jest.genMockFunction();
  });

  it('renders "Pokemons" deep correctly', () => {
    const tree = renderer.create(<Pokemons/>);
    expect(tree).toMatchSnapshot();
  });

  it('renders "Pokemons" shallow correctly', () => {
    const component = shallow(<Pokemons/>);
    expect(component.exists()).toEqual(true);
  });

})
