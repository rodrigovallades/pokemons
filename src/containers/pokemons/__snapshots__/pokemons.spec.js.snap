// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemons container renders "Pokemons" deep correctly 1`] = `
<div>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      <span
        className="badge badge-light"
      >
        0
      </span>
       
      <span>
        Pokemons
      </span>
    </h1>
    <div
      className="pokemons"
    >
      <p
        className="notfound"
      >
        No pokemons found.
      </p>
    </div>
    <div
      className="alert alert-info"
      role="alert"
    >
      <span
        className="octicon octicon-info"
      />
       Tip: Download the 
      <strong>
        <a
          href="https://github.com/zalmoxisus/redux-devtools-extension"
          target="_blnk"
        >
          Redux DevTools
        </a>
      </strong>
       to inspect the Redux store state.
    </div>
  </div>
</div>
`;
